### PR TITLE
Feat/#65 generate tree data

### PR DESCRIPTION
**Pull request by ciaranschutte**
Generates the tree data structure from schema files
Uses two passes of reduce (not most efficient, but most clean right now) to first get mappings, then map. This accounts for parent => child relationships in any order.

Added some basic testing features.

### CODE REVIEW

File: scripts/generateData/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const chalk = require('chalk');
+const get = require('lodash/get');
+
+/**
+ * Generates tree structure for tree viz
+ * Two pass to account for random order
+ * Generates objects in first pass
+ * Maps objects in second pass
+ */
+function generateTreeData(data) {
```
Comment (hlminh2000) [inline scripts/generateData/index.js]: just curious, how is it generating a tree without recursion?

File: scripts/generateData/index.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const singleChildTree = require('./testData/singleChildNodeTree');
+const singleNodeSchema = require('./testData/singleNodeSchema');
+const singleNodeTree = require('./testData/singleNodeTree');
+const nestedChildSchema = require('./testData/nestedChildSchema');
+const nestedChildTree = require('./testData/nestedChildTree');
+const nestedSiblingSchema = require('./testData/nestedSiblingSchema');
+const nestedSiblingTree = require('./testData/nestedSiblingTree');
+
+const writeFile = async (name = 'temp', data) =>
+  await fse.writeJson(`./generateData/temp/${name}.json`, data);
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply scripts/generateData/index.test.js]: Left this in as a convenience . No tests are currently writing to file, no cleanup.
Also added `temp/` to .gitignore

File: scripts/generateData/index.js (hunk 1 above)
Comment (ciaranschutte) [reply scripts/generateData/index.js]: @hlminh2000 
`mapping` holds node objects. 
The `children` property of these objects is manipulated in 2nd reduce.
These object references are passed into `tree`

for example:
-- sample_reg
--- schema
---- donor

would look something like
- `sample_reg` has no parent, so add to root of `tree` (line: 31)
- `specimen` has parent of `sample_reg`, so push to `sample_reg.children` object in mapping, which is referenced in `tree`
- `donor` has parent of `specimen`, pushes to `specimen` mapping, which is already referenced in `tree` under `sample_reg`

Have a recursive version in mind? :)

File: scripts/generateData/index.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hlminh2000) [reply scripts/generateData/index.test.js]: thanks!

File: scripts/generateData/index.js (hunk 1 above)
Comment (hlminh2000) [reply scripts/generateData/index.js]: nope! this is perfect :) would actually avoid recursion myself:)

**Comment by ciaranschutte** [comment]
Thanks @hlminh2000 !

**Pull request merged**